### PR TITLE
[KOGITO-1706] - Updating Architecture documentation

### PR DESCRIPTION
**Pull request by ricardozanini**
See:
https://issues.redhat.com/browse/KOGITO-1706

As we were talking, I asked for you to review this doc. Since nobody replied, I understood that we are ok to go public. Could you please take your time and do a final review?

xoxo

Many thanks for submiting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.MD#sending-a-pull-request)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a unit test that verifies it
- [ ] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

File: docs/ARCHITECTURE.MD [hunk 1]
```diff
@@ -1,72 +1,287 @@
...
+apiVersion: app.kiegroup.org/v1alpha1
+kind: KogitoDataIndex
+metadata:
+  name: data-index
+spec:
+  replicas: 1
+  kafka:
+    useKogitoInfra: true
+  infinispan:
+      useKogitoInfra: true
```
Comment (sutaakar) [inline docs/ARCHITECTURE.MD:100]:
```suggestion
    useKogitoInfra: true
```

File: docs/ARCHITECTURE.MD [hunk 2]
```diff
@@ -1,72 +1,287 @@
...
+    useKogitoInfra: true
+  infinispan:
+      useAuth: true
+      credentials:
+          secretName: infinispan-credentials
+          usernameKey: username
+          passwordKey: password
+      uri: my-infinispan-server:11222
+      saslMechanism: DIGEST-MD5
+      useKogitoInfra: false
```
Comment (sutaakar) [inline docs/ARCHITECTURE.MD:125]:
```suggestion
    useAuth: true
    credentials:
      secretName: infinispan-credentials
      usernameKey: username
      passwordKey: password
    uri: my-infinispan-server:11222
    saslMechanism: DIGEST-MD5
    useKogitoInfra: false
```

File: docs/ARCHITECTURE.MD [hunk 3]
```diff
@@ -1,72 +1,287 @@
...
+
+Another way of achieving local builds is to use the s2i client instead of a Dockerfile. 
+Since building and maintaining Dockerfiles is a well-known approach, it’s preferable over the 
+s2i client.
+
+Since we have the “Kogito CLI” tool that facilitates the integration with the operator, 
+one possible solution is to incorporate the local image build to it. This way, the full circle 
+could be achieved by the tool:
+
+1. Kogito CLI Generates the Kogito project’s skeleton
```
Comment (sutaakar) [inline docs/ARCHITECTURE.MD:260]: Is this going to be provided by Kogito CLI too?
Anyway user can use also Maven archetype to generate the skeleton.

Comment (ricardozanini) [reply docs/ARCHITECTURE.MD:260]: This is just a possibility/we speculating about the future. :)

**Comment by ricardozanini** [comment]
Suggestions applied. @spolti could you please take a look?

**Pull request merged**